### DISCUSSION TITLE
How can I setup this alongside nvim lsp-zero?

### DISCUSSION THREAD

**Discussion by brunobmello25** [simrat39/rust-tools.nvim#405] [Q&A]
I have [lsp zero v1](https://github.com/VonHeikemen/lsp-zero.nvim/tree/v1.x) configured in my neovim and already have all lsp features setup for rust. I was wondering if there is an easy way to setup nvim rust-tools to only use the debugger, or if there is a way to configure it alongside nvim lsp zero

[here](https://github.com/brunobmello25/skeleton.nvim) is my config in case anyone wants to check it out

**Comment by gamezoid123** [comment]
Lsp zero actually lists instructions on how to set up LSP zero along side rust tools. I found it very helpful.

https://github.com/VonHeikemen/lsp-zero.nvim/blob/v2.x/doc/md/guides/quick-recipes.md#setup-with-rust-tools

**Reply by brunobmello25** [reply]
I'm having a bit of trouble setting up keymaps with lsp zero. As you can see from my config [here](https://github.com/brunobmello25/skeleton.nvim/blob/25af3b971d07310139ef511fb9a6cc1eeb779e20/lua/skeleton/plugins/dap.lua#L5-L19) I setup my debugging keymaps as soon as I setup nvim-dap, which means that the continue keymap that is used to start debugging is already set up with dap. Because of that, when I try to debug it says that no configuration is currently found for rust, even though I already have rust-tools configure [here](https://github.com/brunobmello25/skeleton.nvim/blob/25af3b971d07310139ef511fb9a6cc1eeb779e20/lua/skeleton/config/lsp/init.lua#L33-L41)

---

edit: I might have explained it poorly, but I was wondering if there is a way to use the same key bindings and behaviour that I have setup with default nvim-dap with rust-tools, instead of the hover actions menu
